portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,29 +29,52 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<String> greetings = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> greetings = new ArrayList<String>();
-    greetings.add("Hey");
-    greetings.add("Hola");
-    greetings.add("Hello");
-
     // Convert greetings into a json string
     String json = convertToJson(greetings);
 
-    // Send the JSON as the response
+    ArrayList<String> tasks = new ArrayList<>();
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      String commentEx = (String) entity.getProperty("message");
+      tasks.add(commentEx);
+    }
+
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(convertToJson(tasks));
   }
 
   private String convertToJson(ArrayList<String> greetings) {
```

Comment:
It is confusing that the argument name is `greetings` when there is also an instance variable higher up with the same name.